presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/mysql/MySqlFunctionNamespaceManager.java

```diff
@@ -247,12 +247,16 @@ public void dropFunction(QualifiedObjectName functionName, Optional<List<TypeSig
             FunctionNamespaceDao transactionDao = handle.attach(functionNamespaceDaoClass);
             List<SqlInvokedFunction> functions = getSqlFunctions(transactionDao, functionName, parameterTypes);
 
-            checkUnique(functions, functionName);
             checkExists(functions, functionName, parameterTypes);
 
-            SqlInvokedFunction latest = functions.get(0);
-            checkState(latest.getVersion().isPresent(), "Function version missing: %s", latest.getFunctionId());
-            transactionDao.setDeletionStatus(hash(latest.getFunctionId()), latest.getFunctionId(), getLongVersion(latest), true);
+            if (functions.size() > 1) {
```

Comment:
This is assuming that when `functions.size() > 1` user didn't specify parameter list, which should be true if everything is implemented correctly. But It's probably better to check for that (no parameter list provided), and still keep the uniqueness check otherwise? It's not like we'd support `DROP FUNCTION foo(a bigint.*)` do we?

Reply:
- Updated the predicate to `!parameterTypes.isPresent()`
- In case parameter list is specified, I'm now using `Iterables.getOnlyElement` to assert for uniqueness.